SConstruct

```diff
@@ -168,12 +170,17 @@ else:
     json_api_file = os.path.join(os.getcwd(), env["headers_dir"], "extension_api.json")
 
 bindings = env.GenerateBindings(
-    env.Dir("."), [json_api_file, os.path.join(env["headers_dir"], "godot", "gdnative_interface.h")]
+    env.Dir("."), [json_api_file, os.path.join(env["headers_dir"], "godot", "gdnative_interface.h"), "binding_generator.py"]
 )
 
+if(env["cache"]):
+    CacheDir(env["cache_dir"])
+    Decider('MD5')
+
 # Forces bindings regeneration.
 if env["generate_bindings"]:
     AlwaysBuild(bindings)
+    NoCache(bindings)
```

Comment:
Wouldn't this also prevent the files from being cached (so the next build will have no cached bindings)?

Reply:
This indeed prevents generated source from being cached in the same run.
But the next run, if they are up to date, they will be cached.

I don't know how to prevent pulling from the cache without preventing push.
But I think we should not take the risk of pulling generated sources from the cache when explicitly asked for a generation: if, for whatever reason, something not tracked in the dependencies would change the generated sources (Say, an environment variable), this could lead to severe headaches for users not familiar with this!
And the source generation is (on my machine at least?) negligible: I simply wanted to avoid the meaningless recompilations.

I am very open to a solution preventing pull and allowing push :slightly_smiling_face:. I just wanted to stick to the point.

Reply:
Yeah, I guess this is fine, we shouldn't need to specify `generate_bindings=yes` 99% of time anyway.